### PR TITLE
Gus issue 114

### PR DESCRIPTION
**Pull request by zaicheek**
plot_circles function acts as plot_targets but adds color selection as a parameter - test is currently unstable (possibly just animate issues) and z-axis needs correction

### CODE REVIEW

File: tests/test_visualization.py [hunk 1]
```diff
@@ -150,6 +150,29 @@ def test_plot_targets(self):
...
+                [0, 0, 0],
+                [1, 1, 0],
+                [-1, 1, 0],
+                [1, -1, 0],
+                [-1, -1, 0]
+            ])
+            target_radius = 0.1
+            target_color = 'b'
+            plt.figure()
+            plot_circles(target_position, target_radius, (-2, 2, -2, 2))
```
Comment (leoscholl) [inline tests/test_visualization.py:167]: you've got (-2, 2, -2, 2) going into circle_color here, that's why you're getting an error i think

File: aopy/visualization.py [hunk 2]
```diff
@@ -445,6 +445,73 @@ def plot_targets(target_positions, target_radius, bounds=None, alpha=0.5, origin
             ax.set_aspect('equal', adjustable='box')
     if bounds is not None: set_bounds(bounds, ax)
 
+def plot_circles(circle_positions, circle_radius, circle_color = 'b', bounds=None, alpha=0.5, origin=(0, 0, 0), ax=None, unique_only=True):
```
Comment (leoscholl) [inline aopy/visualization.py:448]: i think origin=(0,0,0) is not used

File: aopy/visualization.py [hunk 3]
```diff
@@ -445,6 +445,73 @@ def plot_targets(target_positions, target_radius, bounds=None, alpha=0.5, origin
...
+                [0, 0, 0],
+                [1, 1, 0],
+                [-1, 1, 0],
+                [1, -1, 0],
+                [-1, -1, 0]
+            ])
+            circle_radius = 0.1
+            plot_targets(target_position, circle_radius, (-2, 2, -2, 2))
+
+        .. image:: _images/targets.png
```
Comment (leoscholl) [inline aopy/visualization.py:467]: you can remove this example

File: aopy/visualization.py [hunk 4]
```diff
@@ -445,6 +445,73 @@ def plot_targets(target_positions, target_radius, bounds=None, alpha=0.5, origin
             ax.set_aspect('equal', adjustable='box')
     if bounds is not None: set_bounds(bounds, ax)
 
+def plot_circles(circle_positions, circle_radius, circle_color = 'b', bounds=None, alpha=0.5, origin=(0, 0, 0), ax=None, unique_only=True):    
+    '''
+    Add targets to an axis. If any targets are at the origin, they will appear 
+    in a different color (magenta). Works for 2D and 3D axes
```
Comment (leoscholl) [inline aopy/visualization.py:451]: This docstring is out of date

**Review by leoscholl — COMMENTED** [review]
looks good, i would only ask that you update plot_targets so that it makes use of plot_circles rather than having duplicate code.

File: aopy/visualization.py (hunk 4 above)
Comment (zaicheek) [reply aopy/visualization.py:451]: will create issue to reflect outdated docs in parent/child functions

File: tests/test_visualization.py (hunk 1 above)
Comment (zaicheek) [reply tests/test_visualization.py:167]: good catch, thank you

**Review by zaicheek — COMMENTED** [review]
changes have been effected. plot_circles now draws correctly with color passed as string.

File: aopy/visualization.py [hunk 5]
```diff
@@ -445,6 +445,57 @@ def plot_targets(target_positions, target_radius, bounds=None, alpha=0.5, origin
             ax.set_aspect('equal', adjustable='box')
     if bounds is not None: set_bounds(bounds, ax)
 
+def plot_circles(circle_positions, circle_radius, circle_color = 'b', bounds=None, alpha=0.5, ax=None, unique_only=True):    
+    '''
+    Add targets to an axis. If any targets are at the origin, they will appear 
+    in a different color (magenta). Works for 2D and 3D axes
```
Comment (leoscholl) [inline aopy/visualization.py:451]:
```suggestion
    Add circles to an axis. Works for 2D and 3D axes
```

**Review by leoscholl — APPROVED** [review]

File: aopy/visualization.py (hunk 4 above)
Comment (leoscholl) [reply aopy/visualization.py]: i fixed it.

**Pull request merged**